packages/dashmate/test/e2e/tests/testnetMasternode.spec.js

```diff
@@ -0,0 +1,257 @@
+        containerIds.pop();
+        return containerIds;
+      });
+
+      for (let serviceData of output) {
+        expect(Object.keys(SERVICES)).to.include(serviceData.service);
+        expect(serviceData.status).to.equal('running');
+        expect(listIDs).to.include(serviceData.containerId);
+      }
+    });
+
+    it('Verify status overview before core sync process finish', async () => {
+      const overviewStatus = await dashmate.checkStatus('');
+      const output = JSON.parse(overviewStatus.toString());
+      const coreVersion = core.docker.image.replace(/\/|\(.*?\)|dashpay|dashd:|\-(.*)/g, '');
+
+      expect(output.Network).to.equal(STATUS.test);
+      expect(output['Core Version']).to.equal(coreVersion);
+      expect(output['Core Status']).to.include(ServiceStatusEnum.syncing);
+      expect(output['Masternode Status']).to.equal(STATUS.masternode_status);
+      expect(output['Platform Status']).to.equal(STATUS.platform_status);
+    });
+
+    it('Check host status before core sync process finish', async () => {
+      const hostStatus = await dashmate.checkStatus('host');
+      const output = JSON.parse(hostStatus.toString());
+
+      expect(output.hostname).to.equal(os.hostname());
+      expect(output.uptime).to.include(prettyMs(os.uptime() * 1000, {unitCount: 2}));
+      expect(output.platform).to.equal(os.platform());
+      expect(output.arch).to.equal(os.arch());
+      expect(output.username).to.equal(os.userInfo().username);
+      // expect(output.diskFree).to.equal(0); bugged
+      // expect(output.memory).to.equal(`${prettyByte(os.totalmem())} / ${prettyByte(os.freemem())}`); doesn't work properly on wsl
+      expect(output.cpus).to.equal(os.cpus().length);
+      expect(output.ip).to.equal(await publicIp.v4());
+    });
+
+    it('Check masternode status before core sync process finish', async () => {
+      const masternodeStatus = await dashmate.checkStatus('masternode');
+      const output = JSON.parse(masternodeStatus.toString());
+
+      // expect(output['Masternode status']).to.equal(STATUS.masternode_status); bugged
+      expect(output.sentinel.state).to.equal(STATUS.sentinel_statusNotSynced);
+      });
+
+    it('Restart testnet network before core sync process finish', async () => {
+      const coreStatus = await dashmate.checkStatus('core');
+      const statusBeforeRestart = JSON.parse(coreStatus);
+
+      await dashmate.restart(testnetNetwork);
+
+      await isTestnetServicesRunning(true, container);
+
+      const statusAfterRestart = JSON.parse(coreStatus.toString());
+      if (+statusBeforeRestart.blockHeight !== +statusAfterRestart.blockHeight) {
+        throw new Error('Block height is different after restart.');
+      } else {
+        let blockHeighSync;
+        do {
+          await wait(5000);
+          blockHeighSync = JSON.parse(coreStatus.toString());
+        } while (+blockHeighSync.blockHeight <= +statusAfterRestart.blockHeight);
```

Comment:
You can safely compare without unintended type casting, core status should always return number in this case. If not, then let me fix it))

Reply:
As far as I remember execSync return buffer so toString() method is used for the output at the level of getting core status.